opm/output/eclipse/Summary.cpp

```diff
@@ -1075,6 +1122,59 @@ Summary::Summary( const EclipseState& st,
     }
 }
 
+std::vector< std::pair< std::string, double > >
+efficiency_factors( const smspec_node_type* type,
+                    const Schedule& schedule,
+                    const std::vector< const Well* >& schedule_wells,
+                    size_t timestep ) {
+    std::vector< std::pair< std::string, double > > efac;
+
+    if(    smspec_node_get_var_type(type) != ECL_SMSPEC_GROUP_VAR
+        && smspec_node_get_var_type(type) != ECL_SMSPEC_FIELD_VAR
+        && smspec_node_get_var_type(type) != ECL_SMSPEC_REGION_VAR
+        && !smspec_node_is_total( type ) ) {
+        return efac;
+    }
+
+    const bool is_group = smspec_node_get_var_type(type) == ECL_SMSPEC_GROUP_VAR;
+    const bool is_rate = !smspec_node_is_total( type );
+    const auto &groupTree = schedule.getGroupTree(timestep);
+
+    /* Preferably we would like to use a native validation method ("hasParent")
+     * instead of a try-catch block.
+     * When/if that function is implemented, change the current workflow.*/
```

Comment:
Please mark the function which throws clearly - possibly the `try` block could be a single function call?